packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -702,6 +713,53 @@ class AndroidGradleBuilder implements AndroidBuilder {
       color: TerminalColor.green,
     );
   }
+
+  @override
+  Future<List<String>> getBuildVariants({required FlutterProject project}) async {
+    final Status status = _logger.startProgress(
+      "Running Gradle task 'printBuildVariants'...",
+    );
+    final List<String> command = <String>[
+      _gradleUtils.getExecutable(project),
+      '-q', // suppresses gradle output.
+      'printBuildVariants',
+    ];
+
+    final Stopwatch sw = Stopwatch()
+      ..start();
+    RunResult result;
+    try {
+      result = await _processUtils.run(
+        command,
+        workingDirectory: project.android.hostAppGradleRoot.path,
+        allowReentrantFlutter: true,
+        environment: <String, String>{
+          if (globals.androidSdk?.javaHome != null)
+            'JAVA_HOME': globals.androidSdk!.javaHome!,
+        },
+      );
+    } finally {
+      status.stop();
+    }
+    _usage.sendTiming('print', 'android build variants', sw.elapsed);
+
+    if (result.exitCode != 0) {
+      _logger.printStatus(result.stdout, wrap: false);
+      _logger.printError(result.stderr, wrap: false);
+      throwToolExit(
+        'Gradle task printBuildVariants failed with exit code ${result.exitCode}.',
+        exitCode: result.exitCode,
+      );
+    }
+    final List<String> options = <String>[];
+    for (final String line in result.stdout.split('\n')) {
```

Comment:
Does this work on windows or would the end character in the terminal be different?

Reply:
good catch